backend/src/main/java/ai/verta/modeldb/advancedService/AdvancedServiceImpl.java

```diff
@@ -729,7 +756,21 @@ public void findHydratedProjects(
       // Get the user info from the Context
       var userInfo = authService.getCurrentLoginUserInfo();
 
-      var projectResponse = futureProjectDAO.findProjects(request).get();
+      FindProjects.Response projectResponse;
+      try {
+        projectResponse = futureProjectDAO.findProjects(request).get();
+      } catch (Exception ex) {
+        if (ex.getCause() != null && ex.getCause().getCause() != null) {
+          if (ex.getCause().getCause() instanceof InvalidArgumentException) {
+            throw (InvalidArgumentException) ex.getCause().getCause();
+          } else if (ex.getCause().getCause() instanceof UnimplementedException) {
+            throw (UnimplementedException) ex.getCause().getCause();
+          } else if (ex.getCause().getCause() instanceof PermissionDeniedException) {
+            throw (PermissionDeniedException) ex.getCause().getCause();
+          }
+        }
+        throw ex;
+      }
```

Comment:
In `AdvanceServiceImpl` we do not implemented `InternalFuture` so we have to catch and rethrow errors so it is doing here